odoo/sql_db.py

```diff
@@ -8,77 +8,42 @@
 the ORM does, in fact.
 """
 
-from contextlib import contextmanager
-from datetime import timedelta, datetime
-import itertools
 import logging
 import os
 import time
 import threading
 import uuid
 import warnings
-
+import threading
+import re
+from contextlib import contextmanager
+from datetime import timedelta
 from inspect import currentframe
+
 import psycopg2
 import psycopg2.extras
 import psycopg2.extensions
-from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT, ISOLATION_LEVEL_READ_COMMITTED, ISOLATION_LEVEL_REPEATABLE_READ
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT, ISOLATION_LEVEL_REPEATABLE_READ
 from psycopg2.pool import PoolError
 from psycopg2.sql import SQL, Identifier
 from werkzeug import urls
 
+from . import tools
 from .tools.func import frame_codeinfo, locked
 
+def undecimalize(value, cr):
+    if value is None:
+        return None
+    return float(value)
+
 psycopg2.extensions.register_type(psycopg2.extensions.UNICODE)
+psycopg2.extensions.register_type(psycopg2.extensions.new_type((700, 701, 1700), 'float', undecimalize))
```

Comment:
Not sure movign these here makes things clearer.

Reply:
It is to have all `psycopg2.extensions.register_type` in the same place. If it troubles you, I can remove it?

Reply:
I'm really not sure: this puts both `register_type` calls together but the previous version made it clearer the `register_type` and the callback went together.

Reply:
I see :+1:, I changed it